### PR TITLE
ci(ES-665): stripe ct actions

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
### 🔗 Linked issue

https://alokai.atlassian.net/browse/ES-665

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds actions to build and deploy Stripe CT modules

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: deploy/stripe-ct/action.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: Deploy Storefront
+description: Builds Middleware app docker image
```
Comment (grixu) [inline deploy/stripe-ct/action.yml:2]:
```suggestion
description: Deploy Stripe for Commercetools app
```

File: deploy/stripe-ct/action.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+name: Deploy Storefront
```
Comment (grixu) [inline deploy/stripe-ct/action.yml:1]:
```suggestion
name: Deploy Stripe CT
```

File: deploy/stripe-ct/action.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    - uses: chrnorm/deployment-action@releases/v2
+      name: Create GitHub deployment
+      id: deployment
+      with:
+        token: "${{ github.token }}"
+        environment-url: https://${{ inputs.project_name }}.${{ inputs.cloud_region }}.gcp.storefrontcloud.io
+        environment: production
+        initial-status: in_progress
+
+    - name: Deploy Stripe CT modules (${{ inputs.project_name }}.${{ inputs.cloud_region }}.gcp.storefrontcloud.io)
```
Comment (grixu) [inline deploy/stripe-ct/action.yml:56]: I'm just wondering — WDYT about having this on extra switch in the deploy action?

And second thought — How about middleware is current configuration? For example: I have a unified on CT + stripe - Should I use both action to deploy those apps? Or we should add the middleware here as well?

File: deploy/stripe-ct/action.yml (hunk 1 above)
Comment (grixu) [reply deploy/stripe-ct/action.yml:2]: Or `Deploy Storefront with Stripe for Commercetools` if we gonna add middleware deployment here as well

File: deploy/stripe-ct/action.yml (hunk 2 above)
Comment (grixu) [reply deploy/stripe-ct/action.yml:1]: Or:
```suggestion
name: Deploy Storefront (with Stripe)
```

File: deploy/stripe-ct/action.yml (hunk 1 above)
Comment (grixu) [reply deploy/stripe-ct/action.yml:2]:
```suggestion
description: Deploy Storefront with Stripe for Commercetools
```

File: deploy/action.yml [hunk 4]
```diff
@@ -2,27 +2,27 @@ name: Deploy Storefront
 description: Builds Middleware app docker image
```
Comment (grixu) [inline deploy/action.yml:2]:
```suggestion
description: Deploy the Storefront to the Console
```

If you could make this straight 🙏

File: build-stripe-ct/action.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
+name: Build Stripe CT Module
```
Comment (grixu) [inline build-stripe-ct/action.yml:1]: Please add information about this action in README.md and more detailed description for our SEs here: https://alokai.atlassian.net/wiki/spaces/UST/pages/288227346/Deployment

File: deploy/stripe-ct/action.yml (hunk 2 above)
Comment (grixu) [inline deploy/stripe-ct/action.yml:1]: Please add information about this action in README.md and more detailed description for our SEs here: https://alokai.atlassian.net/wiki/spaces/UST/pages/288227346/Deployment

File: build-stripe-ct/action.yml (hunk 5 above)
Comment (WojtekTheWebDev) [reply build-stripe-ct/action.yml:1]: Done.

File: deploy/stripe-ct/action.yml (hunk 2 above)
Comment (WojtekTheWebDev) [reply deploy/stripe-ct/action.yml]: Done

**Review by Fifciu — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -88,6 +88,36 @@ with:
...
+- `cloud_password`: Cloud API and Docker registry password. Required field.
+- `npm_email`: Email address required for logging into the private NPM registry. Required field.
+- `npm_user`: Username for the private NPM registry. Required field.
+- `npm_pass`: Password for the private NPM registry. Required field.
+- `docker_registry_url`: URL to the Docker image registry. Optional field. Defaults to `registry.vuestorefront.cloud`.
+- `npm_registry`: URL to the private NPM registry. Optional field. Defaults to `https://registrynpm.storefrontcloud.io`.
+- `version`: Version that will be used for docker image tag. Example: 2.3.0, 3.1.0-rc.1. If not passed, github.sha will be used
+- `extension_module_name`: Extension module name. It’s used as the name of the docker image and path where the api is available. Default value is `ct-stripe-extension`.
+- `extension_module_port`: Port of the extension module. Default value is `8080`.
+- `extension_module_config`: Configuration for the extension module.
```
Comment (grixu) [inline README.md:116]:
```suggestion
- `extension_module_config`: Configuration for the extension module. Required field.
```

File: README.md [hunk 7]
```diff
@@ -88,6 +88,36 @@ with:
...
+- `notification_module_name`: Notification module name. It’s used as the name of the docker image and path where the api is available. Default value is `ct-stripe-notification`.
+- `notification_module_port`: Port of the notification module. Default value is `8081`.
+- `notification_module_config`: Configuration for the notification module.
```
Comment (grixu) [inline README.md:119]:
```suggestion
- `notification_module_config`: Configuration for the notification module. Required field.
```

**Review by grixu — APPROVED** [review]
Please, apply those two small suggestions before merge :)

**Pull request merged**